benchmarks/bench_fuzzy_join_vs_others.py

```diff
@@ -0,0 +1,192 @@
+"""
+This benchmark compares the performance of dirty-cat's fuzzy_join compared
+to other fuzzy joining functions available on small toy datasets.
+"""
```

Comment:
Maybe add a conclusion, like the others?
(although their utility could be argued) (For things depending on stuff outside skrub, I would also recommend adding a date, like "As of June 2023, the fuzzy_join compares like...")

Reply:
Great idea, I added the dates. Easier to track this way.